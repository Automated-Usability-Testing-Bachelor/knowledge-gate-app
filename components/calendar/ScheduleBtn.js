import React from "react";
import { View, StyleSheet, TouchableOpacity } from "react-native";
import RedButton from "../RedButton";
import PlusIcon from "./PlusIcon";

const ScheduleBtn = ({ ShowModalCallback }) => {
  return (
    <View style={styles.container}>
      <RedButton
        name="schedule"
        onPress={ShowModalCallback}
        icon={
          <View style={styles.box}>
            <PlusIcon />
          </View>
        }
      />
    </View>
  );
};
const styles = StyleSheet.create({
  container: { paddingHorizontal: 25, marginBottom: 25 },
});

export default ScheduleBtn;
