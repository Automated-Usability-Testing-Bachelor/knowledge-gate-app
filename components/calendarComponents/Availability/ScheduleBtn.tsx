import React from 'react'
import { StyleSheet, View } from 'react-native'
import RedButton from '../../RedButton'
import PlusIcon from '../PlusIcon'

const styles = StyleSheet.create({
  container: { paddingHorizontal: 25, marginBottom: 25 }
})

export type Props = {
  ShowModalCallback: () => void
  testID?: string
}
const ScheduleBtn: React.FC<Props> = ({ ShowModalCallback, testID }) => {
  return (
    <View style={styles.container}>
      <RedButton
        name={'Schedule'}
        onPress={ShowModalCallback}
        testID={testID}
        icon={
          <View>
            <PlusIcon />
          </View>
        }
      />
    </View>
  )
}

export default ScheduleBtn
