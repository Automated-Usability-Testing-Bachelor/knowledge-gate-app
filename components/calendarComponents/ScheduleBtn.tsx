import React from 'react'
import { StyleSheet, View } from 'react-native'
import RedButton from '../RedButton'
import PlusIcon from './PlusIcon'

export type Props = {
  ShowModalCallback: () => void
}
const ScheduleBtn: React.FC<Props> = ({ ShowModalCallback }) => {
  return (
    <View style={styles.container}>
      <RedButton
        name="Schedule"
        onPress={ShowModalCallback}
        icon={
          <View>
            <PlusIcon />
          </View>
        }
      />
    </View>
  )
}
const styles = StyleSheet.create({
  container: { paddingHorizontal: 25, marginBottom: 25 }
})

export default ScheduleBtn
