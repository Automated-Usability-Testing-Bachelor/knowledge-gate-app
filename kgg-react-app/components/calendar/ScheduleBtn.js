import React, { useState } from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import Colors from "../../constants/Colors";
import PlusIcon from "./PlusIcon";

const ScheduleBtn = ({ ShowModalCallback }) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={ShowModalCallback}>
        <View style={styles.box}>
          <PlusIcon />
          <Text style={styles.text}>Schedule</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
};
const styles = StyleSheet.create({
  container: { paddingHorizontal: 25, marginBottom: 25 },
  box: {
    flexDirection: "row",
    alignSelf: "center",
    backgroundColor: Colors.red.color,
    paddingVertical: 8,
    paddingHorizontal: 30,
    borderRadius: 10,
  },
  text: {
    color: "#FFF",
    lineHeight: 18,
  },
});

export default ScheduleBtn;
