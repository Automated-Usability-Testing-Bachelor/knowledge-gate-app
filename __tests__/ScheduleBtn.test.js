/*eslint-disable */

import React from 'react'
import ScheduleBtn from "../components/calendarComponents/Availability/ScheduleBtn"
import { fireEvent, render, waitFor, screen } from '@testing-library/react-native';


// eslint-disable-next-line no-undef
describe('<VisibleItem />', () => {
    const id = "abc"
    const onPress = jest.fn(()=> "pressed");
    it('Test onPress event', async () => {
        
        const activeButton = render(
        <ScheduleBtn ShowModalCallback={onPress} testID="button"  />
        )
        const { getByTestId } = activeButton;
        const button = getByTestId('button');
        expect(activeButton).toBeDefined();

        const activeButtonAsJson = activeButton.toJSON();
        expect(activeButtonAsJson).toMatchSnapshot();
        fireEvent.press(button)
        fireEvent.press(button)
        await waitFor(() => {
            expect(onPress.mock.calls.length).toBe(2)
            expect(onPress.mock.results[0].value).toBe("pressed")
            expect(onPress.mock.results[1].value).toBe("pressed")
        });
      
    });
    it('Test onPress event when not set in props', async () => {
        
        const activeButton = render(
        <ScheduleBtn testID="button"  />
        )
        const { getByTestId } = activeButton;
        const button = getByTestId('button');
        expect(activeButton).toBeDefined();

        const activeButtonAsJson = activeButton.toJSON();
        expect(activeButtonAsJson).toMatchSnapshot();
        fireEvent.press(button)
        fireEvent.press(button)
        await waitFor(() => {
            expect(onPress.mock.calls.length).toBe(0)
        });
      
    });
   
  
});

