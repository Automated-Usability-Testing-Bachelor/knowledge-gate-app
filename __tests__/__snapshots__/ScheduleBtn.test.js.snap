// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VisibleItem /> Test onPress event 1`] = `
<View
  style={
    {
      "marginBottom": 25,
      "paddingHorizontal": 25,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "borderRadius": 10,
          "opacity": 1,
          "padding": 12,
        }
      }
      testID="button"
    >
      <View
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#DD2638",
            "borderRadius": 10,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "lineHeight": 18,
              "paddingHorizontal": 5,
            }
          }
        >
          Schedule
        </Text>
        <View>
          <View
            style={
              {
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              aria-hidden="true"
              bbHeight={18}
              bbWidth={18}
              color={4294967295}
              fill="currentColor"
              focusable={false}
              height={18}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 18,
                    "width": 18,
                  },
                ]
              }
              tintColor={4294967295}
              vbHeight={24}
              vbWidth={24}
              width={18}
            >
              <RNSVGGroup
                fill={
                  [
                    2,
                  ]
                }
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  clipRule={0}
                  d="M12 3.75a.75.75 0 01.75.75v6.75h6.75a.75.75 0 010 1.5h-6.75v6.75a.75.75 0 01-1.5 0v-6.75H4.5a.75.75 0 010-1.5h6.75V4.5a.75.75 0 01.75-.75z"
                  fillRule={0}
                  propList={
                    [
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<VisibleItem /> Test onPress event when not set in props 1`] = `
<View
  style={
    {
      "marginBottom": 25,
      "paddingHorizontal": 25,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "borderRadius": 10,
          "opacity": 1,
          "padding": 12,
        }
      }
      testID="button"
    >
      <View
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#DD2638",
            "borderRadius": 10,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "lineHeight": 18,
              "paddingHorizontal": 5,
            }
          }
        >
          Schedule
        </Text>
        <View>
          <View
            style={
              {
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              aria-hidden="true"
              bbHeight={18}
              bbWidth={18}
              color={4294967295}
              fill="currentColor"
              focusable={false}
              height={18}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 18,
                    "width": 18,
                  },
                ]
              }
              tintColor={4294967295}
              vbHeight={24}
              vbWidth={24}
              width={18}
            >
              <RNSVGGroup
                fill={
                  [
                    2,
                  ]
                }
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  clipRule={0}
                  d="M12 3.75a.75.75 0 01.75.75v6.75h6.75a.75.75 0 010 1.5h-6.75v6.75a.75.75 0 01-1.5 0v-6.75H4.5a.75.75 0 010-1.5h6.75V4.5a.75.75 0 01.75-.75z"
                  fillRule={0}
                  propList={
                    [
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
